### PR TITLE
Fix transaction compatibility issue

### LINKED ISSUE #32: All builds are broken
```
Error in test testUncommittable (Products.Transience.tests.testTransactionHelper.TestTransactionHelper)
Traceback (most recent call last):
  File "/opt/python/2.7.15/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/travis/build/zopefoundation/Products.Sessions/src/Products/Transience/tests/testTransactionHelper.py", line 30, in testUncommittable
    makeTransactionUncommittable(self.t, "test")
  File "/home/travis/build/zopefoundation/Products.Sessions/src/Products/Transience/TransactionHelper.py", line 41, in __init__
    transaction.register(self)
AttributeError: 'Transaction' object has no attribute 'register'
```

https://travis-ci.org/github/zopefoundation/Products.Sessions

@icemac mentioned in the wrap up of the Zope sprint, that this is no easy fix

### PR DESCRIPTION
**Pull request by icemac**
Fixes #32

### CODE REVIEW

File: buildout.cfg [hunk 1]
```diff
@@ -9,6 +9,7 @@ parts =
 
 [versions]
 Products.Sessions =
+transaction = < 3
```
Comment (dataflake) [inline buildout.cfg:12]: This is not needed if you have it in setup.py

**Comment by dataflake** [comment]
Apologies for increasing the size of the patch, I took the opportunity to do some of those cleanups we have been doing in other packages so that the tests have a common configuration and pass.

The version pin in `setup.py` will introduce a version conflict in Zope itself. Maybe it's easier to just fix the incompatibility for real?

**Comment by dataflake** [comment]
@icemac I added a "real" fix in 2f732e5bd52b80ca61ac04318bc689ed09b65769 and think this is ready to go. The `register` call was done in test code only so there is no change in production code behavior at all. Take a look and let me know or merge yourself.

**Comment by icemac** [comment]
@dataflake Thank you for actually fixing the problem. I cannot approve my own PR so I am going to merge it. Should we cut a new release or does it not matter as the change is only in test code?

**Comment by dataflake** [comment]
I'll do a release today after testing the code with the Memcache-based sessioning product I maintain. That one may be affected by transaction 3.x as well. I don't think those guys realized how much would break by removing those APIs. They should have added deprecation warnings with hints what to do instead.

**Comment by dataflake** [comment]
version 4.7 is now released

**Pull request merged**